core/src/scene/sceneLoader.cpp

```diff
@@ -271,6 +273,154 @@ void SceneLoader::loadCameras(Node cameras, View& view) {
 
 }
 
+Filter* SceneLoader::generateFilter(YAML::Node _filter) {
+
+    std::vector<Filter*> filters;
+
+    for(YAML::const_iterator filtItr = _filter.begin(); filtItr != _filter.end(); ++filtItr) {
+
+        Filter* filter;
+
+        if(_filter.IsSequence()) {
+
+            filter = generateFilter(*filtItr);
+
+        } else if(filtItr->first.as<std::string>() == "none") {
+
+            filter = generateNoneFilter(_filter["none"]);
+
+        } else if(filtItr->first.as<std::string>() == "not") {
+
+            filter = generateNoneFilter(_filter["not"]);
+
+        } else if (filtItr->first.as<std::string>() == "any") {
+
+            filter = generateAnyFilter(_filter["any"]);
+
+        } else if (filtItr->first.as<std::string>() == "all") {
+
+            filter = generateFilter(_filter["all"]);
+
+        } else {
+
+            std::string key = filtItr->first.as<std::string>();
+            filter = generatePredicate(_filter[key], key);
+
+        }
+
+        filters.push_back(filter);
+
+    }
+
+    if(filters.size() > 0) {
+        return (new All(filters));
```

Comment:
In the where `filters.size() == 1` we can just return `filters.front()` and save a level of indirection